src/pages/RequestCallPage.js

```diff
@@ -203,27 +203,34 @@ class RequestCallPage extends Component {
      *
      * @returns {Object}
      */
-    getFirstAndLastName({login, displayName}) {
-        let firstName;
-        let lastName;
-
-        if (Str.removeSMSDomain(login) === displayName) {
-            firstName = '';
-            lastName = '';
+    getFirstAndLastName({
+        login,
+        displayName,
+        firstName,
+        lastName,
+    }) {
+        let fName;
+        let lName;
+
+        if (firstName && lastName) {
+            fName = firstName;
+            lName = lastName;
```

Comment:
use early return here

Reply:
@parasharrajat Is this okay?
```
    getFirstAndLastName({
        login,
        displayName,
        firstName,
        lastName,
    }) {
        if (firstName && lastName) {
            return {firstName, lastName};
        }
        if (Str.removeSMSDomain(login) === displayName) {
            return {firstName: '', lastName: ''};
        }

        let fName = '';
        let lName = '';
        const firstSpaceIndex = displayName.indexOf(' ');
        const lastSpaceIndex = displayName.lastIndexOf(' ');

        if (firstSpaceIndex === -1) {
            fName = displayName;
            lName = '';
        } else {
            fName = displayName.substring(0, firstSpaceIndex);
            lName = displayName.substring(lastSpaceIndex);
        }
        return {firstName: fName, lastName: lName};
    }
```

Reply:
yup. But don't forgot to trim the spaces from the parts that you get from displayname.

Reply:
So, our final change will look like this?
```
    getFirstAndLastName({
        login,
        displayName,
        firstName,
        lastName,
    }) {
        if (firstName || lastName) {
            return {firstName: firstName || '', lastName: lastName || ''};
        }
        if (Str.removeSMSDomain(login) === displayName) {
            return {firstName: '', lastName: ''};
        }

        let fName = '';
        let lName = '';
        const firstSpaceIndex = displayName.indexOf(' ');
        const lastSpaceIndex = displayName.lastIndexOf(' ');
        if (firstSpaceIndex === -1) {
            fName = displayName;
            lName = '';
        } else {
            fName = displayName.substring(0, firstSpaceIndex).trim();
            lName = displayName.substring(lastSpaceIndex).trim();
        }
        return {firstName: fName, lastName: lName};
    }
```